### PR TITLE
Add visual only option to Image2D

### PR DESCRIPTION
**Pull request by nabriis**

### CODE REVIEW

File: cuqi/geometry/_geometry.py [hunk 1]
```diff
@@ -437,11 +437,16 @@ class Image2D(Geometry):
         If order = 'C', the image is represented in row-major order.
         if order = 'F', the image is represented column-major order.
 
+    visual_only : bool, Default: False
```
Comment (amal-ghamdi) [inline cuqi/geometry/_geometry.py:440]: Do we need to add a check in CUQIarray when is_par is False 

```python
CUQIarray(np.ones(N,N), is_par=False, geom = geom)
```

where geom is an Image geometry for example.

File: cuqi/geometry/_geometry.py [hunk 2]
```diff
@@ -452,16 +457,30 @@ def par_shape(self):
...
+        # Reshape to image (also for multiple parameter vectors). TODO: #327
+        image = vectors.reshape(self.fun_shape+(-1,), order=self.order) 
+        #Squeeze to return single image if only one parameter vector was given
+        image = image.squeeze()
+        return image
     
     def _plot(self, values, **kwargs):
+        # If only visual, we must convert value to image ourselves
+        if self.visual_only:
+            values = self._vector_to_image(values)
```
Comment (amal-ghamdi) [inline cuqi/geometry/_geometry.py:482]: does this work for multiple images?

File: cuqi/geometry/_geometry.py [hunk 3]
```diff
@@ -452,16 +457,30 @@ def par_shape(self):
...
+        # Else, convert parameter vector into image
+        return self._vector_to_image(pars)
 
     def fun2par(self, funvals):
+        # If geometry is only used for visualization, do nothing
+        if self.visual_only: return funvals
+        # Else, convert image into parameter vector
         return funvals.ravel(order=self.order) #Maybe use reshape((self.dim,), order=self.order)
+
+    def _vector_to_image(self, vectors):
```
Comment (amal-ghamdi) [inline cuqi/geometry/_geometry.py:471]: nice helper function

**Review by amal-ghamdi — APPROVED** [review]
Thank you @nabriis. I think this is very nice option to add. I just have few comments, but I'm approving for now.

File: cuqi/geometry/_geometry.py (hunk 1 above)
Comment (nabriis) [reply cuqi/geometry/_geometry.py:440]: Good point. I realized that we need to switch fun_shape depending on only_visual. Its now updated.

File: cuqi/geometry/_geometry.py (hunk 2 above)
Comment (nabriis) [reply cuqi/geometry/_geometry.py:482]: Yes

**Review by amal-ghamdi — APPROVED** [review]
LGTM!

**Review by jakobsj — APPROVED** [review]

**Pull request merged**